src/instance.rs

```diff
@@ -27,6 +27,16 @@ pub unsafe trait PyNativeType: Sized {
     /// The form of this which is stored inside a `Py<T>` smart pointer.
     type AsRefSource: HasPyGilRef<AsRefTarget = Self>;
 
+    /// Cast `&self` to a `Bound` smart pointer.
+    ///
+    /// This is available as a migration tool to adjust code from the deprecated "GIL Refs"
+    /// API to the `Bound` smart pointer API.
+    fn as_bound<'a, 'py>(self: &'a &'py Self) -> &'a Bound<'py, Self::AsRefSource> {
```

Comment:
Wouldn't the conversion to `Borrowed` be more natural here? Or should we have both?

Reply:
Yes I'll add both. I want to mostly encourage users to use `Bound` as much as possible so that they don't get overwhelmed by choice, and only use `Borrowed` when it's really needed.

Reply:
I was thinking about reducing choice as one can always go from `Borrowed` to `Bound` automatically via `Deref`, i.e. we actually only need `Borrowed` and producing a full `Bound` will just be a call to `.to_owned()` in either approaches.

(I would even consider having `as_bound` return `Borrowed` but that does seem a bit confusing.)

Reply:
Actually I like that idea of reducing choice, as it makes the general pattern much simpler to just have `.as_borrowed()` everywhere. I'll pushed a commit to switch to that.